src/jit/lowerxarch.cpp

```diff
@@ -2603,19 +2649,146 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
+                    // take an actual scalar value, rather than taking a vector and only
+                    // operating on the lower bits.ther than taking a vector and only
+                    // operating on the lower bits.
+
+                    const unsigned expectedSize = genTypeSize(containingNode->gtSIMDBaseType);
+                    const unsigned operandSize  = genTypeSize(node->TypeGet());
+
+                    supportsGeneralLoads = (operandSize == expectedSize);
+
+                    if (!supportsGeneralLoads && ((expectedSize == 1) || (expectedSize == 2)))
+                    {
+                        // The scalar intrinsics that take an 8-bit or 16-bit operand
+                        // should also support general loads when the operand is 32-
+                        // bits in order to handle the case where we have a stack-spilled
+                        // value. This should be safe to do since loads from other 32-
+                        // bit locations (such as fields) should be coming to us in the
+                        // form of a CAST node and will fail the IsContainableMemoryOp check.
+
+                        supportsGeneralLoads = (operandSize == 4);
+                    }
+                    break;
+                }
+
+                case NI_SSE_ConvertScalarToVector128Single:
+                case NI_SSE2_ConvertScalarToVector128Double:
+                case NI_SSE2_ConvertScalarToVector128Int32:
+                case NI_SSE2_ConvertScalarToVector128UInt32:
+                case NI_SSE_X64_ConvertScalarToVector128Single:
+                case NI_SSE2_X64_ConvertScalarToVector128Double:
+                case NI_SSE2_X64_ConvertScalarToVector128Int64:
+                case NI_SSE2_X64_ConvertScalarToVector128UInt64:
+                {
+                    if (!varTypeIsIntegral(node->TypeGet()))
+                    {
+                        // The floating-point overload doesn't require any special semantics
+                        assert(containingIntrinsicId == NI_SSE2_ConvertScalarToVector128Double);
+                        supportsSIMDScalarLoads = true;
+                        supportsGeneralLoads    = supportsSIMDScalarLoads;
+                        break;
+                    }
+
+                    // These intrinsics require the contained operand to be the same
+                    // size as the baseType for the containing node.
+                    //
+                    // If it isn't, we can't mark the node as contained because it
+                    // might avoid a zero or sign-extension that should have otherwise
+                    // occurred.
+                    //
+                    // We need this even though there are overloads that explicitly
+                    // take byte/sbyte/int16/uint16, since the user may be calling
+                    // the int32/uint32 overload with an explicit cast.
```

Comment:
Why does the explicit cast (GT_CAST) get removed?

Reply:
I should probably clarify the comment, the expected size might be 4, and we might get a 1 or 2 byte operand (such as a FIELD) that does not have a cast (since those are implicitly 4 on the operation stack). This was seen in the failure case that lead to the PR.